source/LinkedList.inl

```diff
@@ -135,21 +168,32 @@ template<typename T>
 T&
 DataStructures::LinkedList<T>::front()
 {
+  assert(!empty());
+  return first->datum;
+}
+
+template<typename T>
+const T&
+DataStructures::LinkedList<T>::cfront() const
+{
+  assert(number_of_elements > 0);
```

Comment:
probably best to be consistent about how you do your asserts here. (`number_of_elements > 0` vs `!empty()` above.)

Reply:
ah yeah that's leftover from when `empty()` wasn't `const`-qualified